doc/source/whatsnew/v1.4.0.rst

```diff
@@ -702,6 +702,7 @@ Reshaping
 - Bug in :meth:`DataFrame.append` failing to retain ``index.name`` when appending a list of :class:`Series` objects (:issue:`44109`)
 - Fixed metadata propagation in :meth:`Dataframe.apply` method, consequently fixing the same issue for :meth:`Dataframe.transform`, :meth:`Dataframe.nunique` and :meth:`Dataframe.mode` (:issue:`28283`)
 - Bug in :meth:`DataFrame.stack` with ``ExtensionDtype`` columns incorrectly raising (:issue:`43561`)
+- Fixed joining of :class:`MultiIndex` with overlapping ``IntervalIndex`` levels (:issue:`44096`)
```

Comment:
We are using syntax like: ``Bug in :class:`MultiIndex` with overlapping ``IntervalIndex`` levels raising ...``

Reply:
Adjusted